bin/lib/cli/builds.py

```diff
@@ -0,0 +1,165 @@
+                print("...aborted due to deployment failure!")
+                sys.exit(1)
+        else:
+            old_deploy_staticfiles(branch, to_set)
+        set_current_key(cfg, to_set)
+        if release:
+            print("Marking as a release in sentry...")
+            token = get_ssm_param("/compiler-explorer/sentryAuthToken")
+            result = requests.post(
+                f"https://sentry.io/api/0/organizations/compiler-explorer/releases/{release.version}/deploys/",
+                data=dict(environment=cfg.env.value),
+                headers=dict(Authorization=f'Bearer {token}'))
+            if not result.ok:
+                raise RuntimeError(f"Failed to send to sentry: {result} {result.content.decode('utf-8')}")
+            print("...done", json.loads(result.content.decode()))
+
+
+@builds.command(name="rm_old")
+@click.option('--dry-run/--no-dry-run', help='dry run only')
+@click.argument('max_age', type=int)
+def builds_rm_old(dry_run: bool, max_age: int):
+    """Remove all but the last MAX_AGE builds."""
+    current = get_all_current()
+    max_builds: Dict[str, int] = defaultdict(int)
+    for release in get_releases():
+        max_builds[release.version.source] = max(release.version.number, max_builds[release.version.source])
+    for release in get_releases():
+        if release.key in current:
+            print("Skipping {} as it is a current version".format(release))
+        else:
+            age = max_builds[release.version.source] - release.version.number
+            if age > max_age:
+                if dry_run:
+                    print("Would remove build {}".format(release))
+                else:
+                    print("Removing build {}".format(release))
+                    remove_release(release)
+            else:
+                print("Keeping build {}".format(release))
+
+
+@builds.command(name='list')
+@click.pass_obj
+@click.option('-b', '--branch', type=str, help='show only BRANCH (may be specified more than once)',
+              metavar='BRANCH', multiple=True)
+def builds_list(cfg: Config, branch: Sequence[str]):
+    """List available builds.
+
+    The --> indicates the build currently deployed in this environment."""
+    current = get_current_key(cfg) or ''
+    releases = get_releases()
+    display_releases(current, set(branch), releases)
+
+
+@builds.command(name='history')
+@click.option('--from', 'from_time')
+@click.option('--until', 'until_time')
+@click.pass_obj
+def builds_history(cfg: Config, from_time: Optional[str], until_time: Optional[str]):
+    """Show the history of current versions for this environment."""
+    if from_time is None and until_time is None:
+        if confirm_action(
+                'Do you want list all builds for {}? It might be an expensive operation:'.format(cfg.env.value)):
+            list_all_build_logs(cfg)
+    else:
+        list_period_build_logs(cfg, from_time, until_time)
+
+
+def confirm_branch(release):
```

Comment:
`confirm_branch` and `confirm_action` should probably be moved to the utils file?

for `confirm_branch` also maybe change the parameter to just `branch` and pass `release.branch` at the call site in here.

Reply:
Good idea. will do.